File: backend/Models/Projeto.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class Projeto
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public required string Nome { get; set; }

    [MaxLength(300)]
    public string? Descricao { get; set; }

    public DateTime DataInicio { get; set; }

    public DateTime DataFim { get; set; }

    [Required]
    [MaxLength(150)]
    public required string Programa { get; set; }

    // Relacionamento com Pesquisador
    public int PesquisadorId { get; set; }

    public Pesquisador? Pesquisador { get; set; }
}

File: android/app/src/main/java/com/example/radarfinanceiro/models/Projeto.java
package com.example.radarfinanceiro.models;

public class Projeto {

    private int id;
    private String nome;
    private String descricao;
    private String dataInicio;
    private String dataFim;
    private int duracaoEmDias;
    private String programa;
    private int pesquisadorId;

    public int getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public String getDescricao() {
        return descricao;
    }

    public String getDataInicio() {
        return dataInicio;
    }

    public String getDataFim() {
        return dataFim;
    }

    public int getDuracaoEmDias() {
        return duracaoEmDias;
    }

    public String getPrograma() {
        return programa;
    }

    public int getPesquisadorId() {
        return pesquisadorId;
    }
}
